commit_test.go

```diff
@@ -164,6 +171,41 @@ func TestCommitDecodingWithWhitespace(t *testing.T) {
 	assert.Equal(t, "tree <- initial commit", commit.Message)
 }
 
+func TestCommitDecodingMultilineHeader(t *testing.T) {
+	author := &Signature{Name: "", Email: "john@example.com", When: time.Now()}
+	committer := &Signature{Name: "", Email: "jane@example.com", When: time.Now()}
+
+	treeId := []byte("cccccccccccccccccccc")
+
+	from := new(bytes.Buffer)
+
+	fmt.Fprintf(from, "author %s\n", author)
+	fmt.Fprintf(from, "committer %s\n", committer)
+	fmt.Fprintf(from, "tree %s\n", hex.EncodeToString(treeId))
+	fmt.Fprintf(from, "gpgsig -----BEGIN PGP SIGNATURE-----\n")
+	fmt.Fprintf(from, " <signature>\n")
+	fmt.Fprintf(from, " -----END PGP SIGNATURE-----\n")
+	fmt.Fprintf(from, "\ninitial commit\n")
+
+	flen := from.Len()
+
+	commit := new(Commit)
+	n, err := commit.Decode(from, int64(flen))
+
+	require.Nil(t, err)
+	require.Equal(t, flen, n)
+	require.Len(t, commit.ExtraHeaders, 1)
+
+	hdr := commit.ExtraHeaders[0]
+
+	assert.Equal(t, "gpgsig", hdr.K)
+	assert.EqualValues(t, []string{
+		"-----BEGIN PGP SIGNATURE-----",
+		"<signature>",
+		"-----END PGP SIGNATURE-----"},
+		strings.Split(hdr.V, "\n"))
+}
+
```

Comment:
I think that that's a great idea. I added such a test in 69666fc, which introduces it in `object_db_test.go`, which I think is a more appropriate place for such a test than in `commit_test.go`, since we need to write whole objects, not just the commit serialization.

In fact, this is a such a good idea that it helped me uncover and patch a bug that I found in c2e7ae1. Thanks for looking out :-).